### PR TITLE
feature(event-sourcing) add cache support for aggregates

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description

Add EventStoreManager Cache decorator and aggregate.cache pool

Issue #1267
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [ ] Behat Test

# Checklist:
- [ ] I have read the contribution requirements and fulfill them.
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/event-sourcing/src/Infrastructure/Manager/Decorator/EventStoreManagerCacheDecorator.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public function exists(AggregateId $id): bool
+    {
+        $item = $this->adapter->getItem($id->getValue());
+        if ($item->isHit()) {
+            return true;
+        }
+
+        return $this->manager->exists($id);
```
Comment (piotrkreft) [inline module/event-sourcing/src/Infrastructure/Manager/Decorator/EventStoreManagerCacheDecorator.php:57]: Worth somewhat caching?

File: module/event-sourcing/src/Infrastructure/Manager/Decorator/EventStoreManagerCacheDecorator.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    public function load(AggregateId $id): ?AbstractAggregateRoot
+    {
+        $item = $this->adapter->getItem($id->getValue());
+        if (!$item->isHit()) {
+            $aggregate = $this->manager->load($id);
+            if ($aggregate) {
+                $item->set($aggregate);
+                $this->adapter->save($item);
+            } else {
+                return null;
```
Comment (piotrkreft) [inline module/event-sourcing/src/Infrastructure/Manager/Decorator/EventStoreManagerCacheDecorator.php:37]: I think we still can save the information that the aggregate is not there? Or you think it makes no sense?

**Review by piotrkreft — DISMISSED** [review]

File: module/event-sourcing/src/Infrastructure/Manager/Decorator/EventStoreManagerCacheDecorator.php (hunk 2 above)
Comment (rprzedzik) [reply module/event-sourcing/src/Infrastructure/Manager/Decorator/EventStoreManagerCacheDecorator.php]: fixed

**Review by wfajczyk — DISMISSED** [review]

**Review by piotrkreft — APPROVED** [review]

**Comment by piotrkreft** [comment]
Changelog entry should be added for this one

**Review by piotrkreft — APPROVED** [review]

**Review by wfajczyk — APPROVED** [review]

**Pull request merged**